### PR TITLE
Fix NoMethodError when use trace raise with pattern

### PR DESCRIPTION
**Pull request by st0012**
I also improved tracer's test cases.

### CODE REVIEW

File: lib/debug/tracer.rb [hunk 1]
```diff
@@ -58,14 +58,20 @@ def skip? tp
       if tp.path.start_with?(__dir__) ||
          tp.path.start_with?('<internal:') ||
          ThreadClient.current.management? ||
-         (@pattern && !tp.path.match?(@pattern) && !tp.method_id.match?(@pattern)) ||
```
Comment (st0012) [inline lib/debug/tracer.rb:61]: Exception event's `method_id` could be `nil` and thus cause `NoMethodError` when called with `match?`

**Comment by st0012** [comment]
@ko1 I think `trace raise` should use the exception's name as filtering criteria. I can open another PR to add that if it makes sense to you?

File: lib/debug/tracer.rb (hunk 1 above)
Comment (ko1) [reply lib/debug/tracer.rb:61]: `method_id&.` is enough, and `||` is needed. I'll fix it later.

Comment (st0012) [reply lib/debug/tracer.rb:61]: I missed `||` in refactor 🤦 
I've added https://github.com/ruby/debug/pull/228 for the fix.

**Pull request merged**